### PR TITLE
Add Search Input Events(#110)

### LINKED ISSUE #110: Add new type of search input
Current input types https://taskforce-fe-components.now.sh/?path=/story/search-input--search-input

This component should also support a search input as in https://www.figma.com/file/xdKbESSX397obRR5m9DMM5/CoVid---who-is-who?node-id=0%3A1

### PR DESCRIPTION
**Pull request by schesa**
### What changed?
Added onClick prop for Search Input component
Added onEnter prop for Search Input component
 - Closes #110 

### Actions (optional)
 - [ ] Increase the version of the package if you need to release it after the merge
 - [ ] If you added a new component, please make sure to export it in `../src/index.js`

### CODE REVIEW

File: src/components/search-input/search-input.js [hunk 1]
```diff
@@ -7,30 +7,48 @@ export const SearchInput = ({
   hasIcon = true,
   onValueChange,
   placeholder = "cauta",
-  value
+  value,
+  onClick,
```
Comment (moonflare) [inline src/components/search-input/search-input.js]: I'm not sure we need `onClick`. I wouldn't see a case where we would click on the input field. It's common to have a button next to it if needed, in which case the `onClick` will be on the button and not the input field.

Comment (moonflare) [reply src/components/search-input/search-input.js]: I mean, it doesn't hurt, but it looks like an anti-pattern to me.

Comment (aniri) [reply src/components/search-input/search-input.js]: we do have an icon that can be clicked

Comment (schesa) [reply src/components/search-input/search-input.js]: the onClick is't not actually for the input, it's for the SearchIcon
would it be a good ideea to merge both onClick and onEnter intro one prop "onSearch" ?

Comment (moonflare) [reply src/components/search-input/search-input.js]: You mean the search icon? It's still an anti-pattern. I haven't seen a case where you would click an icon in an input in a long time.

Comment (moonflare) [reply src/components/search-input/search-input.js]: Will get this in the `v1.0.4` release. We can update it later if needed. 🙌

**Pull request merged**